virtualenv.py

```diff
@@ -1376,8 +1379,9 @@ def install_python(home_dir, lib_dir, inc_dir, bin_dir, site_packages, clear, sy
         # the value:
         py_executable = '"%s"' % py_executable
     # NOTE: keep this check as one line, cmd.exe doesn't cope with line breaks
-    cmd = [py_executable, '-c', 'import sys;out=sys.stdout;'
-        'getattr(out, "buffer", out).write(sys.prefix.encode("utf-8"))']
+    cmd = [py_executable, '-c', 'import sys;out=sys.stdout;prefix=sys.prefix;'
+        'prefix=prefix if sys.version_info[0]==2 else prefix.encode(sys.getfilesystemencoding());'
+        'getattr(out, "buffer", out).write(prefix)']
```

Comment:
> Actually my codes implement an "always-bytes protocol". On Python 2, sys.prefix is a bytes-like object, so I print it as-is.

But strings in Python 2 use ```sys.getdefaltencoding()``` as the encoding, not ```sys.getfilesystemencoding()```, so what you're writing is going to have a different encoding depending on the Python version (unless the default and filesystem encodings are the same, which isn't always the case - and on Windows there's also the OEM encoding to make things even more messy).

A "strings encoded in UTF-8" protocol is far more robust. If you've actually found a case where that protocol doesn't work, you don't make it clear what that case is, so I don't see the value in this change.

At the absolute minimum, this code needs a comment explaining what is wrong with the original approach (of passing string values between subprocess and main program using a known encoding) and how the modified version solves that issue. Ideally, there should be a test, but the virtualenv test suite doesn't really go down to this level :-(

Also, you're making the assumption that ```sys.getfilesystemencoding()``` is the same in both the subprocess Python and the current Python. While that's not likely to be violated, I don't see any benefit in taking that risk, when a specific UTF-8 encoding is just as good for converting arbitrary string data to a binary stream, and involves no such risk.